build/ci-pipeline.yml

```diff
@@ -0,0 +1,160 @@
+# DESCRIPTION: 	
+# Builds, tests, and packages the solution for all PR requests.	
+
+name: $(SourceBranchName)-$(Date:yyyyMMdd)$(Rev:-r)
+trigger: none
+
+variables:
+- template: ci-variables.yml
+- template: build-variables.yml
+
+stages:
+- stage: BuildUnitTests
+  displayName: 'Build and run unit tests'
+  dependsOn: []
+  jobs:
+  - job: Windows
+    pool:
+      vmImage: $(WindowsVmImage)
+    steps:
+    - template: ./jobs/build.yml
+
+  - job: Linux
+    pool:
+      vmImage: 'ubuntu-latest'
+    steps:
+    - template: ./jobs/build.yml
+      parameters:
+        packageArtifacts: false
+
+- stage: DockerBuild  
+  displayName: 'Build images'
+  dependsOn: []
+  jobs:
+  - template: ./jobs/docker-build-all.yml
+    parameters: 
+      tag: $(ImageTag)
```

Comment:
Kind of a tangential question. At what point will we assign the `<fhirVersion>:latest` tag to the latest build?

Reply:
The latest tag is a bit of a problem right now. The way that the docker repo works the latest tag is assigned to the image that was made most recently. This could be a CI build, release build, or PR build. The best option I think we have is to disable the latest build, but we haven't talked about this yet.

Reply:
Okay. Did not know that. I think we should update the deployment templates to use `master` as the default value for the `imagetag` parameter instead of using `latest`. So that we can prevent template users from deploying a pr build by mistage.

Reply:
Good idea. I changed it to ```release``` instead of ```master``` and I'll add that tag to the release pipeline, so it always corresponds to the latest release. That way there isn't the thought that the image is our current master branch.

Reply:
If it becomes a problem in future we could also push the PR images to a different registry. Come to think of it, it could also make it easier to clean things up